### PR TITLE
Connection: properly add GET-parameters for the `fetchAuthorizationUrl` API call

### PR DESCRIPTION
**Pull request by sergeymitr**
#### Changes proposed in this Pull Request:
Bug: `fetchAuthorizationUrl` API method (RNA API package) did not include the `no_iframe` parameter in "plain permalinks" mode.
Therefore, site-connected users are unable to connect their user accounts to WP.com if the site has "plain permalinks" enabled.

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
_First, reproduce the issue without checking out the branch._
1. Connect Jetpack in site-only mode.
2. Go to "Settings -> Permalinks" and choose "Plain".
3. Go to Jetpack Dashboard and try to connect the user. You should end up on the `jetpack.authorize_iframe` page with minimal UI.
4. Now checkout the branch, rebuild `plugins/jetpack` if needed, and try to connect the user one more time. Confirm that everything went well.

### CODE REVIEW

File: projects/js-packages/api/index.jsx [hunk 1]
```diff
@@ -92,9 +92,10 @@ function JetpackRestApiClient( root, nonce ) {
 
 		fetchAuthorizationUrl: redirectUri =>
 			getRequest(
-				`${ apiRoot }jetpack/v4/connection/authorize_url?no_iframe=1&redirect_uri=${ encodeURIComponent(
-					redirectUri
-				) }`,
+				addQueryString(
```
Comment (jeherve) [inline projects/js-packages/api/index.jsx:95]: Would it be an option to rely on `addQueryArgs` from `@wordpress/url` here?

**Comment by leogermani** [comment]
Don't you think it's time to deprecate the `jetpack_use_iframe_authorization_flow` filter and never set `$iframe_flow` to true again?

https://github.com/Automattic/jetpack/blob/e9f949811b11d3b1a9cfd80b5b7b284d9f46afc1/projects/packages/connection/src/class-manager.php#L986

This would ensure the iframe flow will never be returned in any case.

File: projects/js-packages/api/index.jsx (hunk 1 above)
Comment (sergeymitr) [reply projects/js-packages/api/index.jsx:95]: Good point, I'll rewrite that.

**Comment by sergeymitr** [comment]
Hi @leogermani.

> Don't you think it's time to deprecate the jetpack_use_iframe_authorization_flow filter and never set $iframe_flow to true again?

Deprecate - yes, good idea, I'll do that.
But I don't think we can remove that filter entirely, as it will break older Jetpacks. We should wait a bit longer.

**Comment by leogermani** [comment]
if you deprecate it and force it to false, no need to enforce the `no_iframe` parameter in the API request

**Comment by sergeymitr** [comment]
@leogermani,
I'm worried that if we disable that in the package, old JS might receive the non-iframe URL where it expects the iframe one, breaking everything.
This is an unlikely situation where one needs to have an old Jetpack installed alongside a fresh plugin with latest Connection package. But we need to take that into account still.
As far as I know, we cannot force the non-iframe flow over old Jetpack versions. We should be able to for the main flow, but not for the secondary ones.

If I'm missing something, please let me know.

File: projects/js-packages/api/index.jsx (hunk 1 above)
Comment (sergeymitr) [reply projects/js-packages/api/index.jsx]: Done 👍

**Review by samiff — APPROVED** [review]
Looks good, after patching I no longer saw the `jetpack.authorize_iframe`

**Review by samiff — APPROVED** [review]

**Comment by sergeymitr** [comment]
r235541-wpcom

**Pull request merged**